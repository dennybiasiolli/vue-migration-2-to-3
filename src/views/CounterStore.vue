<template>
  <div>
    <p>Count is: {{ count }}</p>
    <p>Number is {{ isEven ? 'even' : 'odd' }}</p>
    <button @click="increment">Increment</button>
  </div>
</template>

<script setup>
import { computed } from 'vue'
import { useStore } from 'vuex'

const store = useStore()

const count = computed(() => store.state.count)

const isEven = computed(() => store.getters.isEven)

const increment = () => store.dispatch('increment')
</script>
