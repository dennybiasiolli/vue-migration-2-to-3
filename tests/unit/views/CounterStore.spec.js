import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import CounterStore from '@/views/CounterStore.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('CounterStore.vue', () => {
  let store
  let actions

  beforeEach(() => {
    actions = {
      increment: jest.fn(),
    }
    store = new Vuex.Store({
      state: { count: 0 },
      getters: {
        isEven: () => true,
      },
      actions,
    })
  })

  it('renders as expected', () => {
    const wrapper = shallowMount(CounterStore, { store, localVue })
    expect(wrapper).toMatchSnapshot()
  })

  it('should call increment after button click', () => {
    const wrapper = shallowMount(CounterStore, { store, localVue })
    wrapper.find('button').trigger('click')
    expect(actions.increment).toHaveBeenCalled()
  })
})
