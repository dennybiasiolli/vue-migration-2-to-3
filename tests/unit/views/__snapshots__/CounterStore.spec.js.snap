// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterStore.vue renders as expected 1`] = `
<div>
  <p>Count is: 0</p>
  <p>Number is even</p> <button>Increment</button>
</div>
`;
